updated with changes

yoyounik opened the pull request:
## Describe your changes

Changes made:

Validation Logic Update:
validation.js: Updated nameSchema to allow non-ASCII letters and spaces.
Truncation for Long Names:
Sidebar/index.css: Added CSS rules to truncate long names in the user context menu.

## Issue number

Issue Number : https://github.com/bluewave-labs/Checkmate/issues/1561#issuecomment-2593847431

## Please ensure all items are checked off before requesting a review:

- [x] I deployed the application locally.
- [x] I have performed a self-review and testing of my code.
- [x] I have included the issue # in the PR.
- [x] I have labelled the PR correctly.
- [x] The issue I am working on is assigned to me.
- [x] I didn't use any hardcoded values (otherwise it will not scale, and will make it difficult to maintain consistency across the application).
- [x] I made sure font sizes, color choices etc are all referenced from the theme.
- [x] My PR is granular and targeted to one specific feature.
- [x] I took a screenshot or a video and attached to this PR if there is a UI change.

gorkem-bwl commented:
Can you add a video here? I don't see it here or in the comments.

Also the regex does not allow special characters like hyphens (e.g., "Anne-Marie"), apostrophes (e.g., "O'Connor").

yoyounik commented:
@gorkem-bwl ,
While both the frontend and backend code are ready, I'm running into some issues with Docker that's preventing the backend from running smoothly on my local machine.
I've tried several different approaches to resolve the issue, but it is not running properly.
i am trying to reach out to @ajhollid to help resolve this issue as soon as possible.
If you could give any peace of advice to run this, it would be really helpful for me too.

in detail problem:
I've tried several different approaches to resolve the issue, including restarting services, checking configurations, and ensuring the correct containers are running. Despite these efforts, the Docker daemon has been unresponsive, and I’m unable to get the Redis container to connect properly.

gorkem-bwl commented:
No worries. Let's see if it works on our end on a live system, and get back to you on this then. Next time, please do not click on a check item if you haven't completed it (e.g "I took a screenshot or a video and attached to this PR if there is a UI change").

ajhollid commented:
@yoyounik you've got two PRs open for this issue now, is the older one redundant?  If so, please close it and rename this one appropriately.  Thank you!

ajhollid commented:
> @gorkem-bwl , While both the frontend and backend code are ready, I'm running into some issues with Docker that's preventing the backend from running smoothly on my local machine. I've tried several different approaches to resolve the issue, but it is not running properly. i am trying to reach out to @ajhollid to help resolve this issue as soon as possible. If you could give any peace of advice to run this, it would be really helpful for me too.
> 
> in detail problem: I've tried several different approaches to resolve the issue, including restarting services, checking configurations, and ensuring the correct containers are running. Despite these efforts, the Docker daemon has been unresponsive, and I’m unable to get the Redis container to connect properly.

@yoyounik have you declared the required environmental variables?  Please post your client and server .env files so we can inspect and make sure they are setup correctly.

yoyounik commented:
@ajhollid the client side is running fine. The only problem i am facing is starting off with server, i have no redis installed and i guess that is why the server is not starting off.

yoyounik commented:
@ajhollid i will try installing redis and testing it once, and then i will let you know in a while .

yoyounik commented:
@ajhollid , the server is running now on port 5000,:
![image](https://github.com/user-attachments/assets/aefeb7b4-6006-460c-9854-45f688df8ba6)

the only problem is that , it is showing something like this , when running it on web:
![image](https://github.com/user-attachments/assets/6fdfd0af-10a3-4a3e-add9-5082fcbb3d15)

ajhollid commented:
The server does not accept get requests on /

I imagine you want to connect to he Client, not the server.

Client runs on port 5173

yoyounik commented:
yes @ajhollid 😊
in client .env file , i added this:
VITE_APP_API_BASE_URL="http://localhost:5000/api/v1" 

and in server .env file , i added this:

CLIENT_HOST="http://localhost:5173"
JWT_SECRET="my_secret"
DB_TYPE="MongoDB"
DB_CONNECTION_STRING="mongodb://localhost:27017/uptime_db"
REDIS_HOST="127.0.0.1"
REDIS_PORT=6379
TOKEN_TTL="99d"
PAGESPEED_API_KEY=<api_key>
SYSTEM_EMAIL_HOST="smtp.gmail.com"
SYSTEM_EMAIL_PORT=465
SYSTEM_EMAIL_ADDRESS=<email_address>
SYSTEM_EMAIL_PASSWORD=<password>
REFRESH_TOKEN_SECRET="my_refresh"
REFRESH_TOKEN_TTL="99d"


Can you give any hints where and how to run it, i have tried to run it in many ways

ajhollid commented:
@yoyounik go to the client directory and run `npm run dev`

yoyounik commented:
yes @ajhollid , after going to client and running npm run dev:
and after logging in, it is showing like this:

![image](https://github.com/user-attachments/assets/103b101e-d194-42dc-a14e-ae800d5416a9)

ajhollid commented:
> yes @ajhollid , after going to client and running npm run dev: and after logging in, it is showing like this:
> 
> ![image](https://private-user-images.githubusercontent.com/124652765/406529293-103b101e-d194-42dc-a14e-ae800d5416a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc3Mzk3MzksIm5iZiI6MTczNzczOTQzOSwicGF0aCI6Ii8xMjQ2NTI3NjUvNDA2NTI5MjkzLTEwM2IxMDFlLWQxOTQtNDJkYy1hMTRlLWFlODAwZDU0MTZhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyNFQxNzIzNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zODA3OGJjMjVjYjc4OWI0ZmNkMDExMWNkNGQxMmFkOWQzZDE3NTI0Mjg1ZDY0NjNhZWVhOTkwYzcwNjI2NGQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.klG125R4i1tDWY_h3veizuNs-X7wzvLtloJYlYmw9SI)

It appears your Client cannot talk to your server.

I suggest checking your Server logs and seeing what is going on.

My guess would be Mongo and Redis images are not running.

If that is the case, I suggest you [closely follow the steps set out in the documentation](https://bluewavelabs.gitbook.io/checkmate/users-guide/quickstart#dev-quickstart) 

to properly start your application.

ajhollid commented:
Closing as inactive
